AxeWP/wp-graphql-headless-login
# feat: Add support for Cookie configuration and `logout` mutation

## What
This PR enhances the loginCookie functionality by introducing a new AuthCookie class. This class adds options for configuring the cookie's expiration and SameSite attribute. This should resolve this [issue](https://github.com/AxeWP/wp-graphql-headless-login/issues/104).

## Why
Currently, the authentication cookie being set is session-only. This means that when a user closes their browser, they need to re-authenticate upon reopening. This change allows for persistent authentication sessions, reducing the need for frequent logins.

## How
This PR introduces the AuthCookie class which replaces the default behavior for setting authentication cookies. The AuthCookie class:

Provides the option to set a custom expiration time for the authentication cookie.
Supports the SameSite attribute for cookies, which enhances security by controlling how cookies are sent with cross-site requests.
Allows users to set the cookie as persistent if they opt for the "remember me" functionality, which is currently not supported by default.

## Testing Instructions
Login using a Login provider. Verify cookies and check that SameSite is set to None.

## Additional Info
Things to improve: Add options in admin for samesite configuration & domain cookie.

## Checklist:

- [x] I included the relevant changes in CHANGELOG.md
